### PR TITLE
OCI Vault

### PR DESCRIPTION
**Pull request by anders-swanson**
Adds oci vault as a module

### CODE REVIEW

File: spring-cloud-oci-autoconfigure/src/main/java/com/oracle/cloud/spring/vault/VaultAutoConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+public class VaultAutoConfiguration {
+    private final VaultProperties properties;
+
+    public VaultAutoConfiguration(VaultProperties properties) {
+        this.properties = properties;
+    }
+
+    @Bean
+    @RefreshScope
+    @ConditionalOnProperty(name = "spring.cloud.oci.genai.embedding.enabled", havingValue = "true", matchIfMissing = true)
```
Comment (andytael) [inline spring-cloud-oci-autoconfigure/src/main/java/com/oracle/cloud/spring/vault/VaultAutoConfiguration.java:47]: spring.cloud.oci.genai.embedding.enabled is probably wrong

File: spring-cloud-oci-vault/src/main/java/com/oracle/cloud/spring/vault/VaultImpl.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import com.oracle.bmc.vault.model.UpdateSecretDetails;
+import com.oracle.bmc.vault.requests.CreateSecretRequest;
+import com.oracle.bmc.vault.requests.ScheduleSecretDeletionRequest;
+import com.oracle.bmc.vault.requests.UpdateSecretRequest;
+import com.oracle.bmc.vault.responses.CreateSecretResponse;
+import com.oracle.bmc.vault.responses.ScheduleSecretDeletionResponse;
+import com.oracle.bmc.vault.responses.UpdateSecretResponse;
+import org.springframework.util.Assert;
+
+public class VaultImpl implements Vault {
```
Comment (andytael) [inline spring-cloud-oci-vault/src/main/java/com/oracle/cloud/spring/vault/VaultImpl.java:24]: Not sure if it is on purpose, but I'm messing DeleteSecret and ListSecrets

File: spring-cloud-oci-autoconfigure/src/main/java/com/oracle/cloud/spring/vault/VaultAutoConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    @ConditionalOnMissingBean(Vault.class)
+    public Vault embeddingModel(Vaults vaults, Secrets secrets) {
```
Comment (andytael) [inline spring-cloud-oci-autoconfigure/src/main/java/com/oracle/cloud/spring/vault/VaultAutoConfiguration.java:49]: embeddingModel?

Comment (markxnelson) [reply spring-cloud-oci-autoconfigure/src/main/java/com/oracle/cloud/spring/vault/VaultAutoConfiguration.java:49]: would vault() be a better name?

File: spring-cloud-oci-autoconfigure/src/main/java/com/oracle/cloud/spring/vault/VaultProperties.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package com.oracle.cloud.spring.vault;
```
Comment (markxnelson) [inline spring-cloud-oci-autoconfigure/src/main/java/com/oracle/cloud/spring/vault/VaultProperties.java:1]: check the copyright notices, applies to several files

File: spring-cloud-oci-samples/spring-cloud-oci-vault-sample/src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2024, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+
+spring.cloud.oci.region.static=us-chicago-1
+spring.cloud.oci.config.type=file
+
+spring.cloud.oci.vault.compartment=${OCI_COMPARTMENT_ID}
+spring.cloud.oci.vault.vault-id=${OCI_VAULT_ID}
+spring.cloud.oci.vault.enabled=true
```
Comment (markxnelson) [inline spring-cloud-oci-samples/spring-cloud-oci-vault-sample/src/main/resources/application.properties:9]: nitpick, i think the autoconfig will define this one, so its not strictly needed.  no harm having it though

File: spring-cloud-oci-vault/src/main/java/com/oracle/cloud/spring/vault/Vault.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import com.oracle.bmc.secrets.model.SecretBundleContentDetails;
+import com.oracle.bmc.secrets.responses.GetSecretBundleByNameResponse;
+import com.oracle.bmc.vault.model.CreateSecretDetails;
+import com.oracle.bmc.vault.model.UpdateSecretDetails;
+import com.oracle.bmc.vault.responses.CreateSecretResponse;
+import com.oracle.bmc.vault.responses.ScheduleSecretDeletionResponse;
+import com.oracle.bmc.vault.responses.UpdateSecretResponse;
+
+public interface Vault {
+    GetSecretBundleByNameResponse getSecret(String secretName);
```
Comment (markxnelson) [inline spring-cloud-oci-vault/src/main/java/com/oracle/cloud/spring/vault/Vault.java:21]: probably should add some javadoc, i think the other modules have it

**Review by markxnelson — COMMENTED** [review]
few small comments

File: spring-cloud-oci-autoconfigure/src/main/java/com/oracle/cloud/spring/vault/VaultAutoConfiguration.java (hunk 1 above)
Comment (andytael) [reply spring-cloud-oci-autoconfigure/src/main/java/com/oracle/cloud/spring/vault/VaultAutoConfiguration.java:47]: Correct, should be vault

**Review by markxnelson — APPROVED** [review]

**Review by markxnelson — APPROVED** [review]

**Pull request merged**